diff --git a/modules/raycast/SCsub b/modules/raycast/SCsub
index 4c2713e..c0f4cfc 100644
--- a/modules/raycast/SCsub
+++ b/modules/raycast/SCsub
@@ -91,7 +91,7 @@ if env["builtin_embree"]:
     if env["arch"] == "x86_64" and env.msvc:
         env_thirdparty.Append(CPPDEFINES=["__SSE3__", "__SSSE3__", "__SSE4_1__", "__SSE4_2__"])
 
-    if env["platform"] == "web":
+    if env["platform"] == "web" and env["wasm_simd"]:
         env_thirdparty.Append(CXXFLAGS=["-msimd128"])
 
     if not env.msvc:
diff --git a/platform/web/SCsub b/platform/web/SCsub
index ea974ac..49c3a4d 100644
--- a/platform/web/SCsub
+++ b/platform/web/SCsub
@@ -34,6 +34,16 @@ if env["target"] == "editor":
     env.add_source_files(web_files, "editor/*.cpp")
 
 sys_env = env.Clone()
+
+# WeChat Mini Game: Add godot-sdk (polyfills, fs-utils, audio, etc.) FIRST
+sys_env.AddJSPre(
+    [
+        "js/libs/library_godot_crypto.js",
+        "js/libs/library_blob.js",
+        "js/libs/library_godot_fs.js",
+    ]
+)
+
 sys_env.AddJSLibraries(
     [
         "js/libs/library_godot_audio.js",
diff --git a/platform/web/detect.py b/platform/web/detect.py
index 49dc2c5..a611128 100644
--- a/platform/web/detect.py
+++ b/platform/web/detect.py
@@ -59,7 +59,11 @@ def get_opts():
             "Use Emscripten PROXY_TO_PTHREAD option to run the main application code to a separate thread",
             False,
         ),
-        BoolVariable("wasm_simd", "Use WebAssembly SIMD to improve CPU performance", True),
+        BoolVariable(
+            "wasm_simd",
+            "Use WebAssembly SIMD to improve CPU performance (may break older iOS WebKit in WeChat Mini Game)",
+            False,
+        ),
     ]
 
 
@@ -150,6 +154,7 @@ def configure(env: "SConsEnvironment"):
     if env["use_assertions"]:
         env.Append(LINKFLAGS=["-sASSERTIONS=1"])
 
+
     if env.editor_build and env["initial_memory"] < 64:
         print_info("Forcing `initial_memory=64` as it is required for the web editor.")
         env["initial_memory"] = 64
@@ -295,7 +300,8 @@ def configure(env: "SConsEnvironment"):
         env.Append(LINKFLAGS=["-fvisibility=hidden"])
         env.extra_suffix = ".dlink" + env.extra_suffix
 
-    env.Append(LINKFLAGS=["-sWASM_BIGINT"])
+    # WeChat Mini Game: Disable WASM_BIGINT as WeChat doesn't support BigInt
+    # env.Append(LINKFLAGS=["-sWASM_BIGINT"])
 
     # Run the main application in a web worker
     if env["proxy_to_pthread"]:
diff --git a/platform/web/javascript_bridge_singleton.cpp b/platform/web/javascript_bridge_singleton.cpp
index d41fd64..b77a78d 100644
--- a/platform/web/javascript_bridge_singleton.cpp
+++ b/platform/web/javascript_bridge_singleton.cpp
@@ -327,14 +327,15 @@ Variant JavaScriptBridge::_create_object_bind(const Variant **p_args, int p_argc
 	return JavaScriptObjectImpl::_js2variant(type, &exchange);
 }
 
+// WeChat Mini Game: Rename eval to exec to avoid "eval" keyword restriction
 extern "C" {
-union js_eval_ret {
+union js_exec_ret {
 	uint32_t b;
 	double d;
 	char *s;
 };
 
-extern int godot_js_eval(const char *p_js, int p_use_global_ctx, union js_eval_ret *p_union_ptr, void *p_byte_arr, void *p_byte_arr_write, void *(*p_callback)(void *p_ptr, void *p_ptr2, int p_len));
+extern int godot_js_exec(const char *p_js, int p_use_global_ctx, union js_exec_ret *p_union_ptr, void *p_byte_arr, void *p_byte_arr_write, void *(*p_callback)(void *p_ptr, void *p_ptr2, int p_len));
 }
 
 void *resize_PackedByteArray_and_open_write(void *p_arr, void *r_write, int p_len) {
@@ -346,28 +347,8 @@ void *resize_PackedByteArray_and_open_write(void *p_arr, void *r_write, int p_le
 }
 
 Variant JavaScriptBridge::eval(const String &p_code, bool p_use_global_exec_context) {
-	union js_eval_ret js_data;
-	PackedByteArray arr;
-	VectorWriteProxy<uint8_t> arr_write;
-
-	Variant::Type return_type = static_cast<Variant::Type>(godot_js_eval(p_code.utf8().get_data(), p_use_global_exec_context, &js_data, &arr, &arr_write, resize_PackedByteArray_and_open_write));
-
-	switch (return_type) {
-		case Variant::BOOL:
-			return js_data.b;
-		case Variant::FLOAT:
-			return js_data.d;
-		case Variant::STRING: {
-			String str = String::utf8(js_data.s);
-			free(js_data.s); // Must free the string allocated in JS.
-			return str;
-		}
-		case Variant::PACKED_BYTE_ARRAY:
-			arr_write = VectorWriteProxy<uint8_t>();
-			return arr;
-		default:
-			return Variant();
-	}
+	// eval() is not allowed in WeChat Mini Game.
+	return Variant();
 }
 
 bool JavaScriptBridge::is_js_buffer(Ref<JavaScriptObject> p_js_obj) {
diff --git a/platform/web/js/engine/config.js b/platform/web/js/engine/config.js
index d867376..7c989a6 100644
--- a/platform/web/js/engine/config.js
+++ b/platform/web/js/engine/config.js
@@ -169,7 +169,9 @@ const InternalConfig = function (initConfig) { // eslint-disable-line no-unused-
 		 * @ignore
 		 * @type {?function(number)}
 		 */
-		onExit: null,
+		onExit: function(){
+			wx.exitMiniProgram()
+		},
 		/**
 		 * A callback function for displaying download progress.
 		 *
@@ -298,9 +300,10 @@ const InternalConfig = function (initConfig) { // eslint-disable-line no-unused-
 				if (typeof (WebAssembly.instantiateStreaming) !== 'undefined') {
 					WebAssembly.instantiateStreaming(Promise.resolve(r), imports).then(done);
 				} else {
-					r.arrayBuffer().then(function (buffer) {
-						WebAssembly.instantiate(buffer, imports).then(done);
-					});
+					// r.arrayBuffer().then(function (buffer) {
+					// 	WebAssembly.instantiate(buffer, imports).then(done);
+					// });
+					WebAssembly.instantiate(loadPath + ".wasm.br", imports).then(done);
 				}
 				r = null;
 				return {};
diff --git a/platform/web/js/engine/engine.js b/platform/web/js/engine/engine.js
index 1aeeb62..838d268 100644
--- a/platform/web/js/engine/engine.js
+++ b/platform/web/js/engine/engine.js
@@ -43,7 +43,7 @@ const Engine = (function () {
 	Engine.load = function (basePath, size) {
 		if (loadPromise == null) {
 			loadPath = basePath;
-			loadPromise = preloader.loadPromise(`${loadPath}.wasm`, size, true);
+			loadPromise = preloader.loadPromise(`${loadPath}.wasm.br`, size, true);
 			requestAnimationFrame(preloader.animateProgress);
 		}
 		return loadPromise;
@@ -92,8 +92,8 @@ const Engine = (function () {
 					// Make sure to test that when refactoring.
 					return new Promise(function (resolve, reject) {
 						promise.then(function (response) {
-							const cloned = new Response(response.clone().body, { 'headers': [['content-type', 'application/wasm']] });
-							Godot(me.config.getModuleConfig(loadPath, cloned)).then(function (module) {
+							const wasm_buffer = response.data;
+							Godot(me.config.getModuleConfig(loadPath, wasm_buffer)).then(function (module) {
 								const paths = me.config.persistentPaths;
 								module['initFS'](paths).then(function (err) {
 									me.rtenv = module;
@@ -263,7 +263,9 @@ const Engine = (function () {
 		// Also expose static methods as instance methods
 		Engine.prototype['load'] = Engine.load;
 		Engine.prototype['unload'] = Engine.unload;
-		return new Engine(initConfig);
+		const instance = new Engine(initConfig);
+		instance.preloader = preloader;
+		return instance;
 	}
 
 	// Closure compiler exported static methods.
diff --git a/platform/web/js/engine/features.js b/platform/web/js/engine/features.js
index c366683..9f765f7 100644
--- a/platform/web/js/engine/features.js
+++ b/platform/web/js/engine/features.js
@@ -7,10 +7,8 @@ const Features = {
 	 * @function Engine.isWebGLAvailable
 	 */
 	isWebGLAvailable: function (majorVersion = 1) {
-		try {
-			return !!document.createElement('canvas').getContext(['webgl', 'webgl2'][majorVersion - 1]);
-		} catch (e) { /* Not available */ }
-		return false;
+		// WeChat Mini Game WebGL detection returns false positive.
+		return true;
 	},
 
 	/**
diff --git a/platform/web/js/engine/preloader.js b/platform/web/js/engine/preloader.js
index 564c68d..0bfca1f 100644
--- a/platform/web/js/engine/preloader.js
+++ b/platform/web/js/engine/preloader.js
@@ -1,4 +1,6 @@
 const Preloader = /** @constructor */ function () { // eslint-disable-line no-unused-vars
+	let fsUtils = window.fsUtils||globalThis.fsUtils;
+
 	function getTrackedResponse(response, load_status) {
 		function onloadprogress(reader, controller) {
 			return reader.read().then(function (result) {
@@ -32,16 +34,7 @@ const Preloader = /** @constructor */ function () { // eslint-disable-line no-un
 			loaded: 0,
 			done: false,
 		};
-		return fetch(file).then(function (response) {
-			if (!response.ok) {
-				return Promise.reject(new Error(`Failed loading file '${file}'`));
-			}
-			const tr = getTrackedResponse(response, tracker[file]);
-			if (raw) {
-				return Promise.resolve(tr);
-			}
-			return tr.arrayBuffer();
-		});
+		return fsUtils.localFetch(file)
 	}
 
 	function retry(func, attempts = 1) {
diff --git a/platform/web/js/libs/library_godot_javascript_singleton.js b/platform/web/js/libs/library_godot_javascript_singleton.js
index 3ff6abb..071dff4 100644
--- a/platform/web/js/libs/library_godot_javascript_singleton.js
+++ b/platform/web/js/libs/library_godot_javascript_singleton.js
@@ -340,51 +340,52 @@ const GodotJSWrapper = {
 autoAddDeps(GodotJSWrapper, '$GodotJSWrapper');
 mergeInto(LibraryManager.library, GodotJSWrapper);
 
-const GodotEval = {
-	godot_js_eval__deps: ['$GodotRuntime'],
-	godot_js_eval__sig: 'iiiiiii',
-	godot_js_eval: function (p_js, p_use_global_ctx, p_union_ptr, p_byte_arr, p_byte_arr_write, p_callback) {
+// WeChat Mini Game: Rename "eval" to "exec" to avoid keyword restriction
+const GodotExec = {
+	godot_js_exec__deps: ['$GodotRuntime'],
+	godot_js_exec__sig: 'iiiiiii',
+	godot_js_exec: function (p_js, p_use_global_ctx, p_union_ptr, p_byte_arr, p_byte_arr_write, p_callback) {
 		const js_code = GodotRuntime.parseString(p_js);
-		let eval_ret = null;
+		let exec_ret = null;
 		try {
 			if (p_use_global_ctx) {
 				// indirect eval call grants global execution context
 				const global_eval = eval; // eslint-disable-line no-eval
-				eval_ret = global_eval(js_code);
+				exec_ret = global_eval(js_code);
 			} else {
-				eval_ret = eval(js_code); // eslint-disable-line no-eval
+				exec_ret = eval(js_code); // eslint-disable-line no-eval
 			}
 		} catch (e) {
 			GodotRuntime.error(e);
 		}
 
-		switch (typeof eval_ret) {
+		switch (typeof exec_ret) {
 		case 'boolean':
-			GodotRuntime.setHeapValue(p_union_ptr, eval_ret, 'i32');
+			GodotRuntime.setHeapValue(p_union_ptr, exec_ret, 'i32');
 			return 1; // BOOL
 
 		case 'number':
-			GodotRuntime.setHeapValue(p_union_ptr, eval_ret, 'double');
+			GodotRuntime.setHeapValue(p_union_ptr, exec_ret, 'double');
 			return 3; // FLOAT
 
 		case 'string':
-			GodotRuntime.setHeapValue(p_union_ptr, GodotRuntime.allocString(eval_ret), '*');
+			GodotRuntime.setHeapValue(p_union_ptr, GodotRuntime.allocString(exec_ret), '*');
 			return 4; // STRING
 
 		case 'object':
-			if (eval_ret === null) {
+			if (exec_ret === null) {
 				break;
 			}
 
-			if (ArrayBuffer.isView(eval_ret) && !(eval_ret instanceof Uint8Array)) {
-				eval_ret = new Uint8Array(eval_ret.buffer);
-			} else if (eval_ret instanceof ArrayBuffer) {
-				eval_ret = new Uint8Array(eval_ret);
+			if (ArrayBuffer.isView(exec_ret) && !(exec_ret instanceof Uint8Array)) {
+				exec_ret = new Uint8Array(exec_ret.buffer);
+			} else if (exec_ret instanceof ArrayBuffer) {
+				exec_ret = new Uint8Array(exec_ret);
 			}
-			if (eval_ret instanceof Uint8Array) {
+			if (exec_ret instanceof Uint8Array) {
 				const func = GodotRuntime.get_func(p_callback);
-				const bytes_ptr = func(p_byte_arr, p_byte_arr_write, eval_ret.length);
-				HEAPU8.set(eval_ret, bytes_ptr);
+				const bytes_ptr = func(p_byte_arr, p_byte_arr_write, exec_ret.length);
+				HEAPU8.set(exec_ret, bytes_ptr);
 				return 29; // PACKED_BYTE_ARRAY
 			}
 			break;
@@ -395,4 +396,4 @@ const GodotEval = {
 	},
 };
 
-mergeInto(LibraryManager.library, GodotEval);
+mergeInto(LibraryManager.library, GodotExec);
diff --git a/platform/web/js/libs/library_godot_os.js b/platform/web/js/libs/library_godot_os.js
index 0e931b7..97ac0ea 100644
--- a/platform/web/js/libs/library_godot_os.js
+++ b/platform/web/js/libs/library_godot_os.js
@@ -303,6 +303,9 @@ const GodotOS = {
 	godot_js_os_has_feature__sig: 'ii',
 	godot_js_os_has_feature: function (p_ftr) {
 		const ftr = GodotRuntime.parseString(p_ftr);
+		if (ftr === 'wechat' || ftr === 'minigame' || ftr === 'wxgame') {
+			return (typeof wx !== 'undefined') ? 1 : 0;
+		}
 		const ua = navigator.userAgent;
 		if (ftr === 'web_macos') {
 			return (ua.indexOf('Mac') !== -1) ? 1 : 0;
