diff --git a/platform/web/emscripten_helpers.py b/platform/web/emscripten_helpers.py
index 6f3c051..f73f0d3 100644
--- a/platform/web/emscripten_helpers.py
+++ b/platform/web/emscripten_helpers.py
@@ -38,12 +38,14 @@ def create_template_zip(env, js, wasm, side):
         wasm,
         "#platform/web/js/libs/audio.worklet.js",
         "#platform/web/js/libs/audio.position.worklet.js",
+        "#platform/web/js/libs/audio.worker.js",
     ]
     out_files = [
         zip_dir.File(binary_name + ".js"),
         zip_dir.File(binary_name + ".wasm"),
         zip_dir.File(binary_name + ".audio.worklet.js"),
         zip_dir.File(binary_name + ".audio.position.worklet.js"),
+        zip_dir.File(binary_name + ".audio.worker.js"),
     ]
     # Dynamic linking (extensions) specific.
     if env["dlink_enabled"]:
